queue/abstract/queue_session.lua

```diff
@@ -0,0 +1,195 @@
+        if cur_time >= exp_time then
+            if queue_session._on_session_remove ~= nil then
+                queue_session._on_session_remove(session_uuid)
+            end
+            inactive_sessions[session_uuid] = nil
+        end
+    end
+end
+
+--- Create an expiration fiber to cleanup expired sessions.
+local function create_expiration_fiber()
+    local exp_fiber = fiber.create(function()
+        while true do
+            if box.info.ro == false then
+                local status, err = pcall(cleanup_inactive_sessions)
+                if status == false then
+                    log.error('An error occurred while cleanup the sessions: %s',
+                        err)
+                end
+            end
+            fiber.sleep(1)
+        end
+    end)
+    exp_fiber:name('queue_expiration_fiber')
+
+    return exp_fiber
+end
+
+--- Identify the session.
+-- If session_uuid == nil - create a new UUID for the connection
+-- and return it, otherwise connect to the existing one.
+local function identify(conn_id, session_uuid)
+    local queue_session_ids = box.space._queue_session_ids
+    local session_ids = queue_session_ids.index.box_sid:get(conn_id)
+    local cur_uuid = session_ids and session_ids[2]
+
+    if session_uuid == nil and cur_uuid == nil then
+        -- Generate new UUID for the session.
+        cur_uuid = uuid.bin()
+        queue_session_ids:insert{conn_id, cur_uuid}
+    elseif session_uuid ~= nil then
+        -- Validate UUID.
+        if not pcall(uuid.frombin, session_uuid) then
+            error('Invalid UUID format.')
+        end
+
+        -- identify using a previously created session.
+        -- Check that a session with this uuid exists.
+        local ids_by_uuid = queue_session_ids.index.uuid:select(
+            session_uuid, { limit = 1 })[1]
+        if ids_by_uuid == nil and inactive_sessions[session_uuid] == nil then
+            error('The UUID ' .. uuid.frombin(session_uuid):str() ..
+                ' is unknown.')
+        end
+
+        if cur_uuid ~= session_uuid then
+            if cur_uuid ~= nil then
+                queue_session.remove_connection(conn_id)
+            end
+            queue_session_ids:insert({conn_id, session_uuid})
+            cur_uuid = session_uuid
+        end
+    end
+
+    -- Exclude the session from inactive.
+    inactive_sessions[cur_uuid] = nil
+
+    return cur_uuid
+end
+
+local function on_session_remove(callback)
+    if type(callback) ~= 'function' then
+        error('The "on_session_remove" argument type must be "function"')
+    end
+    queue_session._on_session_remove = callback
+end
+
+--- Remove a connection from the list of active connections and
+-- release its tasks if necessary.
+local function remove_connection(conn_id)
+    local queue_session_ids = box.space._queue_session_ids
+    local session_uuid = queue_session.get_uuid(conn_id)
+
+    queue_session_ids:delete{conn_id}
```

Comment:
Jusr raw idea (maybe for further PRs): it may be helpful to have ability to trace connections and sessions and issue a log entry (on verbose or even debug level) when a connection is added or removed from a session or moved to another session.

Reply:
Sounds good.